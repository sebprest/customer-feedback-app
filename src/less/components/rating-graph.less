.rating-graph {
  height: 100px;
}
