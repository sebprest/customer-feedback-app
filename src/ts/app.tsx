import React from "react";

const App: React.FunctionComponent = React.memo(() => <h1>Hello</h1>);

export default App;
