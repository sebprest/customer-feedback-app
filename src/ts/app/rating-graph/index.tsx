import { Line, LineChart, ResponsiveContainer, YAxis } from "recharts";
import React from "react";
import { connect } from "react-redux";

import { ReviewsState } from "^/common/types";
import { StoreState } from "^/store/types";

interface StateProps {
  reviews: ReviewsState;
}

type Props = StateProps;

export const RatingGraph: React.FunctionComponent<Props> = React.memo(
  (props) => {
    return (
      <ResponsiveContainer>
        <LineChart data={props.reviews} margin={{ top: 5 }}>
          <YAxis />
          <Line type="monotone" dataKey={"rating"} animationDuration={0} />
        </LineChart>
      </ResponsiveContainer>
    );
  }
);

const mapStateToProps = ({ reviews }: StoreState): StateProps => {
  return {
    reviews,
  };
};

export default connect(mapStateToProps)(RatingGraph);
