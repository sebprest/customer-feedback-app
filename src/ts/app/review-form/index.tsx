import React, { useState } from "react";
import { connect } from "react-redux";

import { addReview } from "./actions";
import ReviewFormFields from "./review-form-fields";

type Props = {
  addReview: typeof addReview;
};

export const ReviewForm: React.FunctionComponent<Props> = React.memo(
  (props) => {
    const defaultValues = { name: "", email: "", rating: 0, comment: "" };
    const [values, setValues] = useState(defaultValues);

    const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
      const targetName = event.target.name;
      const targetValue = event.target.value;

      setValues((values) => ({ ...values, [targetName]: targetValue }));
    };

    const onAddReview = (event: React.FormEvent<HTMLFormElement>) => {
      event.preventDefault();
      props.addReview(values);
      setValues(() => defaultValues);
    };

    return (
      <ReviewFormFields
        addReview={onAddReview}
        updateFormState={handleInputChange}
        formState={values}
      />
    );
  }
);

export default connect(null, { addReview })(ReviewForm);
