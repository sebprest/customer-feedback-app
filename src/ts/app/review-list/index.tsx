import React from "react";
import { connect } from "react-redux";

import { ReviewsState } from "^/common/types";
import { StoreState } from "^/store/types";
import Review from "./review";

interface StateProps {
  reviews: ReviewsState;
}

type Props = StateProps;

export const ReviewList: React.FunctionComponent<Props> = React.memo(
  (props) => (
    <div className="review-list">
      <h2>All reviews ({props.reviews.length}):</h2>
      {props.reviews.map((review, index) => (
        <Review review={review} key={index} />
      ))}
    </div>
  )
);

const mapStateToProps = ({ reviews }: StoreState): StateProps => {
  return {
    reviews,
  };
};

export default connect(mapStateToProps)(ReviewList);
