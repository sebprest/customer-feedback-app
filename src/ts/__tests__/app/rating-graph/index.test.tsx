jest.mock("recharts", () => ({
  YAxis: (props: unknown) => <div {...props}>YAxis</div>,
  Line: (props: unknown) => <div {...props}>Line</div>,
  LineChart: (props: unknown) => <div {...props}>LineChart</div>,
  ResponsiveContainer: (props: unknown) => (
    <div {...props}>ResponsiveContainer</div>
  ),
}));
import React from "react";
import renderer from "react-test-renderer";

import { FAKE_REVIEW } from "^/__tests__/__helpers__/fake-review";
import { RatingGraph } from "^/app/rating-graph";

describe("RatingGraph", () => {
  it("should render a graph displaying data about the provided reviews", () => {
    const ratingGraph = renderer.create(
      <RatingGraph reviews={[FAKE_REVIEW]} />
    );

    expect(ratingGraph).toMatchSnapshot();
  });
});
