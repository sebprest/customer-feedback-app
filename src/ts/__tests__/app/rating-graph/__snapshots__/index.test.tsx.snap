// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingGraph should render a graph displaying data about the provided reviews 1`] = `
<div>
  ResponsiveContainer
</div>
`;
